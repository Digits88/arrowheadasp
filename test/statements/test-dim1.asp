Test Dim statement:

Dim VarA, VarB(2), VarC(3, 4), VarD(5, 6, 7)
	<% Dim VarA, VarB(2), VarC(3, 4), VarD(5, 6, 7) %><br>

VarA = <%= VarA %><br>

UBound(VarB) = <%= UBound(VarB) %><br>

UBound(VarC, 1) = <%= UBound(VarC, 1) %><br>
UBound(VarC, 2) = <%= UBound(VarC, 2) %><br>

UBound(VarD, 1) = <%= UBound(VarD, 1) %><br>
UBound(VarD, 2) = <%= UBound(VarD, 2) %><br>
UBound(VarD, 3) = <%= UBound(VarD, 3) %><br>

Sub TestSetANoDim <% Sub TestSetANoDim %><br>
  VarA = "Hello" <% VarA = "Hello" %><br>
  VarA = <%= VarA %><br>
End Sub <% End Sub %><br>

TestSetANoDim <% TestSetANoDim %><br>

VarA = <%= VarA %><br>

Sub TestSetADim <% Sub TestSetADim %><br>
  Dim VarA <% Dim VarA %><br>
  VarA = "Greetings" <% VarA = "Greetings" %><br>
  VarA = <%= VarA %><br>
End Sub <% End Sub %><br>

TestSetADim <% TestSetADim %><br>

VarA = <%= VarA %><br>
