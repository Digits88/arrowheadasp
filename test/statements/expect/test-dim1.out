Test Dim statement:

Dim VarA, VarB(2), VarC(3, 4), VarD(5, 6, 7)
	<br>

VarA = <br>

UBound(VarB) = 2<br>

UBound(VarC, 1) = 3<br>
UBound(VarC, 2) = 4<br>

UBound(VarD, 1) = 5<br>
UBound(VarD, 2) = 6<br>
UBound(VarD, 3) = 7<br>

Sub TestSetANoDim <br>

TestSetANoDim <br>
  VarA = "Hello" <br>
  VarA = Hello<br>
End Sub <br>

VarA = Hello<br>

Sub TestSetADim <br>

TestSetADim <br>
  Dim VarA <br>
  VarA = "Greetings" <br>
  VarA = Greetings<br>
End Sub <br>

VarA = Hello<br>
